openstack/blockstorage/v3/volumetypes/results.go

```diff
@@ -0,0 +1,81 @@
+package volumetypes
+
+import (
+	"encoding/json"
+
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/pagination"
+)
+
+// Volume Type contains all the information associated with an OpenStack Volume Type.
+type VolumeType struct {
+	// Unique identifier for the volume type.
+	ID string `json:"id"`
+	// Human-readable display name for the volume type.
+	Name string `json:"name"`
+	// Human-readable description for the volume type.
+	Description string `json:"description"`
+	// Arbitrary key-value pairs defined by the user.
+	ExtraSpecs map[string]string `json:"extra_specs"`
+	// Whether the volume type is publicly visible.
+	IsPublic bool `json:"is_public"`
+}
+
+// UnmarshalJSON another unmarshalling function
+func (r *VolumeType) UnmarshalJSON(b []byte) error {
+	type tmp VolumeType
+	var s struct {
+		tmp
+	}
+	err := json.Unmarshal(b, &s)
+	if err != nil {
+		return err
+	}
+
+	*r = VolumeType(s.tmp)
+	return nil
+}
```

Comment:
This method doesn't seem to be required here. A custom `UnmarshalJSON` function is only to smooth/convert types. For example, converting a time field to `time.Time` or if a result is returned both as a `int` and a `string`.